libraries/BLE/src/BLEUtils.cpp

```diff
@@ -0,0 +1,190 @@
+
+#include "ArduinoBLE.h"
+#include "BLEUtils.h"
+#include "internal/ble_client.h"
+
+String BLEUtils::macAddressBT2String(const bt_addr_le_t &bd_addr)
+{
+    char mac_string[BT_ADDR_STR_LEN];
+    snprintf(mac_string, BT_ADDR_STR_LEN, "%2.2X:%2.2X:%2.2X:%2.2X:%2.2X:%2.2X",
+            bd_addr.val[5], bd_addr.val[4], bd_addr.val[3],
+            bd_addr.val[2], bd_addr.val[1], bd_addr.val[0]);
+    String temp(mac_string);
+    return temp;
+}
+
+void BLEUtils::macAddressString2BT(const char* mac_str, bt_addr_le_t &bd_addr)
+{
+    char temp[] = {0, 0, 0};
+    int strLength = strlen(mac_str);
+    int length = 0;
+    
+    bd_addr.type = BT_ADDR_LE_PUBLIC;
+
+    for (int i = strLength - 1; i >= 0 && length < MAX_UUID_SIZE; i -= 2)
+    {
+        if (mac_str[i] == ':') 
+        {
+            i++;
+            continue;
+        }
+
+        temp[0] = mac_str[i - 1];
+        temp[1] = mac_str[i];
+
+        bd_addr.val[length] = strtoul(temp, NULL, 16);
```

Comment:
the array `val` has a size of 6, and it seems that I could cause an overrun (and possibly a crash) by simply passing in a long-enough string (if the string is long enough, then the loop will continue until `length == MAX_UUID_SIZE`, which happens to be 16, not 6).